packages/components/src/Button.tsx

```diff
@@ -31,6 +31,8 @@ interface BaseButtonProps extends React.ComponentPropsWithRef<'button'> {
   icon?: IconDefinition | JSX.Element;
   active?: boolean;
   'data-testid'?: string;
+  'data-index'?: number;
+  'data-operator'?: string;
```

Comment:
We shouldn't add `data-index` and `data-operator` attributes (which are quite specific to the `AdvancedFilterCreator`).
Since `data-*` attributes are common (see MDN article: https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes), and the names can be whatever are necessary for the context.
I _think_ this is possible using Mapped Types: https://www.typescriptlang.org/docs/handbook/2/mapped-types.html
See this [TypeScript playground here](https://www.typescriptlang.org/play?jsx=0#code/C4TwDgpgBAhgJnACgJwgMwJYA8AqB7AeQCMArCAY2AB4cAaKRKCLYCAOzgGcpPhkM2AcwB8UALxQA3gFgAUFCgBtANJQBUANYQQeNFByxuq5qw7de-IVAD8UAAYASSYgC+T5S7tQAXFDYQANwhkAF1fHBUQuRc5OQB6OP1waABVYAwAG25AMGpAeD-Y2QFWZDQYcmgCMHS8Nk4AQSk5BQU2GABbCF8LAUEAbmiCouDS8qhK6tqAIUb5ZrAYZGBOXzYAVzaiYP7ZGNk5IZKyiqqMGs4AYRrMQRnmpS0QLr4esLGTs4aAH1gEFHRsfDEMiUKjjU5TegAcgAApDhNtdnJyGdgFA2iBLmxrr4wWdMddxFIXNt4okAOp4ZAaZZydH4jCCAB0MGAfAAjITJK0Or5IdyIJDibSMVcGczWcgAEycgBE0Pmi04Mt8bKFsjpoqZLL4AGZZfKFktlVA2fR+bz+YKSbIElAAKLIZCU8LJKCQyRQBVLFbrTbIXpQFyQtTcNh4VEwTicBmtIgZaDAPBQUCQN242pfH5IVCYXCEUgUajpziTehymXCSGM4X0rUSgAssq9SpVarkQA) where a prop key is prefixed with `@`.
We essentially want to do the same thing but for keys prefixed with `data-`, and value is always a `string`.

Then we use a spread operator to apply them to the `<button>` tag.

Reply:
Data attributes are kind of anti-React I think. You can create any attribute you want as a prop in React, so we should be leaning on those instead unless there's a specific reason it needs to be a data attribute

Also, I saw there's an aria attribute added the the prop types, but all valid aria props are already part of the type since it extends the react component type.